src/shared_modules/utils/socketDBWrapper.hpp

```diff
@@ -164,7 +185,25 @@ class SocketDBWrapper final
 
         if (!m_exceptionStr.empty())
         {
-            throw std::runtime_error(m_exceptionStr);
+            switch (m_queryStatus)
+            {
+            case DbQueryStatus::JSON_PARSING:
+                throw std::runtime_error(m_exceptionStr);
+            case DbQueryStatus::EMPTY_RESPONSE:
+                throw SocketDbWrapperException(m_exceptionStr);
+            case DbQueryStatus::QUERY_ERROR:
+                throw SocketDbWrapperException(m_exceptionStr);
+            case DbQueryStatus::QUERY_IGNORE:
+                throw SocketDbWrapperException(m_exceptionStr);
+            case DbQueryStatus::QUERY_UNKNOWN:
+                throw SocketDbWrapperException(m_exceptionStr);
+            case DbQueryStatus::QUERY_NOT_SYNCED:
+                throw SocketDbWrapperException(m_exceptionStr);
+            case DbQueryStatus::INVALID_RESPONSE:
+                throw std::runtime_error(m_exceptionStr);
+            default:
+                throw std::runtime_error(m_exceptionStr);
```

Comment:
This can be simplified
```suggestion
            case DbQueryStatus::EMPTY_RESPONSE:
            case DbQueryStatus::QUERY_ERROR:
            case DbQueryStatus::QUERY_IGNORE:
            case DbQueryStatus::QUERY_UNKNOWN:
            case DbQueryStatus::QUERY_NOT_SYNCED:
                throw SocketDbWrapperException(m_exceptionStr);
            case DbQueryStatus::JSON_PARSING:
            case DbQueryStatus::INVALID_RESPONSE:
            default:
                throw std::runtime_error(m_exceptionStr);
```

Also, add the warning and error logging to this part of the code